src/include/miopen/inline_vector.hpp

```diff
@@ -0,0 +1,274 @@
+    {
+        if(real_size > N)
+        {
+            MIOPEN_THROW("Input data size is bigger than InlineVector's capacity");
+        }
+
+        if(!std::is_scalar_v<T>)
+        {
+            MIOPEN_THROW("InlineVector support only scalar values for now");
+        }
+
+        std::copy(first, last, _data.begin());
+    }
+
+    // Copy/move operator
+    InlineVector& operator=(const InlineVector& inline_vec)     = default;
+    InlineVector& operator=(InlineVector&& inline_vec) noexcept = default;
+
+    // Iterators
+    T* begin() noexcept { return _data.begin(); }
+
+    const T* begin() const noexcept { return _data.begin(); }
+
+    T* end() noexcept { return (_data.begin() + real_size); }
+
+    const T* end() const noexcept { return (_data.begin() + real_size); }
+
+    // Constant iterator
+    const T* cbegin() const noexcept { return _data.cbegin(); }
+
+    const T* cend() const noexcept { return (_data.cbegin() + real_size); }
+
+    // Reverse iterators
+    std::reverse_iterator<T*> rbegin() noexcept { return std::reverse_iterator<T*>(end()); }
+
+    std::reverse_iterator<const T*> rbegin() const noexcept
+    {
+        return std::reverse_iterator<const T*>(end());
+    }
+
+    std::reverse_iterator<T*> rend() noexcept { return std::reverse_iterator<T*>(begin()); }
+
+    std::reverse_iterator<const T*> rend() const noexcept
+    {
+        return std::reverse_iterator<const T*>(begin());
+    }
+
+    // Constant reverse iterators
+    std::reverse_iterator<const T*> crbegin() const noexcept
+    {
+        return std::reverse_iterator<const T*>(cend());
+    }
+
+    std::reverse_iterator<const T*> crend() const noexcept
+    {
+        return std::reverse_iterator<const T*>(cbegin());
+    }
+
+    // Element access
+    T& operator[](std::size_t n) noexcept { return _data[n]; }
+
+    const T& operator[](std::size_t n) const noexcept { return _data[n]; }
+
+    // Element access with boundaries check
+    T& at(std::size_t n)
+    {
+        if(n >= real_size)
+        {
+            MIOPEN_THROW("Access to InlineVector is out of range");
+        }
+        return _data.at(n);
+    }
+
+    const T& at(std::size_t n) const
+    {
+        if(n >= real_size)
+        {
+            MIOPEN_THROW("Access to InlineVector is out of range");
+        }
+        return _data.at(n);
+    }
+
+    // Access to first element
+    T& front()
+    {
+        if(empty())
+        {
+            MIOPEN_THROW("Cannot get front element, InlineVector is empty");
+        }
+        return (*begin());
+    }
+
+    const T& front() const
+    {
+        if(empty())
+        {
+            MIOPEN_THROW("Cannot get front element, InlineVector is empty");
+        }
+        return (*begin());
+    }
+
+    // Access to last element
+    T& back()
+    {
+        if(empty())
+        {
+            MIOPEN_THROW("Cannot get back element, InlineVector is empty");
+        }
+        return *(end() - 1);
```

Comment:
```suggestion
        return *std::prev(end());;
```

Usually front/back functions do not check ranges. but probably it's ok.